xla/stream_executor/cuda/cuda_blas_lt.cc

```diff
@@ -401,23 +401,25 @@ absl::Status BlasLt::MatmulPlan::ValidateInputs(
 absl::Status BlasLt::MatmulPlan::DoMatmul(
     Stream* stream, const void* alpha, DeviceMemoryBase a, DeviceMemoryBase b,
     const void* beta, DeviceMemoryBase c, DeviceMemoryBase d,
-    const MatmulAlgorithm& algorithm, ScratchAllocator& scratch_allocator,
```

Comment:
Could you possible do the stream_executor changes in a backward-compatible way? Tensorflow is also using these methods, so better would be to create new methods that are used from XLA, then once that has been merged, update Tensorflow and remove the old methods.

Reply:
I did the fix that keep both APIs of allocating through scratch allocator, or through pre-allocated workspace buffer.